### PR TITLE
feat: add localized indicator to field label

### PR DESCRIPTION
**Pull request by JessRynkar**
On any localized field, appends `locale` on the end of the label.

### CODE REVIEW

**Review by DanRibbens — CHANGES_REQUESTED** [review]
I am looking at the `test/localizations` and clicking around. It doesn't appear that all the labels are impacted by this change.
Should a localized block show the current locale?
image

File: packages/ui/src/fields/FieldLabel/index.tsx [hunk 1]
```diff
@@ -58,10 +61,11 @@ export const FieldLabel: FieldLabelClientComponent = (props) => {
           ? props.label
           : props?.field && 'label' in props.field && (props.field.label as StaticLabel) // type assertion needed for `row` fields
       }
+      localized={props?.field && 'localized' in props.field ? props.field.localized : false}
       required={
         typeof props.required !== 'undefined'
           ? props.required
-          : props?.field && 'required' in props.field && (props.field?.required as boolean) // type assertion needed for `group` fields
+          : props?.field && 'required' in props.field && (props.field?.required) // type assertion needed for `group` fields
```
[the remark this answers is not in the document]
Comment (JessChowdhury) [reply packages/ui/src/fields/FieldLabel/index.tsx]: Ah nope, this happened when I saved the file - reverted.

**Comment by JessChowdhury** [comment]
> I am looking at the `test/localizations` and clicking around. It doesn't appear that all the labels are impacted by this change. Should a localized block show the current locale? image

Yes, will correct this.

File: packages/ui/src/fields/FieldLabel/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (DanRibbens) [reply packages/ui/src/fields/FieldLabel/index.tsx]: If eslint removed it and this builds, I'd say we can drop it and the comment.

**Comment by DanRibbens** [comment]
I need clarification on this. I was expecting that the label at the top most localized field would be the one to show the indicator.
Now I see on your branch we are showing the `text` field has the label `en`. Instead though I was expecting that the Content at the top level would have the indicator because that is the level at which we have added localization.

image

**Comment by JessChowdhury** [comment]
> I need clarification on this. I was expecting that the label at the top most localized field would be the one to show the indicator. Now I see on your branch we are showing the `text` field has the label `en`. Instead though I was expecting that the Content at the top level would have the indicator because that is the level at which we have added localization.
> 
> image

Ah okay we might need to talk through this - I thought we have situations where the block could be localized but have a field with `localized: false` within in, in which case you would want to know on a field by field basis. Might be a better way to do this...

**Comment by DanRibbens** [comment]
> I thought we have situations where the block could be localized but have a field with localized: false within

No, we just need to have the indicator shown at the label for any field that has `localization: true` and then any subfields need to be ignored. Even if the fields had `localized: true` or `false` nested inside another `localized: true` it should be ignored.

**Comment by JessChowdhury** [comment]
> No, we just need to have the indicator shown at the label for any field that has `localization: true` and then any subfields need to be ignored.

Gotcha - I have updated it so the indicator only appends fields that have `localization: true`.

File: packages/ui/src/fields/FieldLabel/index.tsx [hunk 2]
```diff
@@ -26,14 +28,15 @@ const DefaultFieldLabel: React.FC<GenericLabelProps> = (props) => {
...
-
   const { i18n } = useTranslation()
+  const { code } = useLocale()
 
   if (labelFromProps) {
     return (
       <Element className={`field-label ${unstyled ? 'unstyled' : ''}`} htmlFor={htmlFor}>
         {getTranslation(labelFromProps, i18n)}
         {required && !unstyled && <span className="required">*</span>}
+        {localized && <span className="localized">{code}</span>}
```
Comment (DanRibbens) [inline packages/ui/src/fields/FieldLabel/index.tsx:39]: Looks like were going with en dash, also the code should only be used when there isn't a `locale.label`

**Review by DanRibbens — CHANGES_REQUESTED** [review]
Well shit, sorry I missed in the indicator design in the earier reviews.

Also it looks like we're missing the indicator for checkbox field types.

One more round!

**Review by denolfe — APPROVED** [review]

**Pull request merged**